lib/ansible/modules/cloud/amazon/efs.py

```diff
@@ -407,8 +407,8 @@ def converge_file_system(self, name, tags, targets):
             changed = filter(
                 lambda sid: not targets_equal(['SubnetId', 'IpAddress', 'NetworkInterfaceId'],
                                               current_targets[sid], targets[sid]), intersection)
-            targets_to_delete = list(targets_to_delete) + changed
-            targets_to_create = list(targets_to_create) + changed
+            targets_to_delete = list(targets_to_delete) + list(changed)
```

Comment:
Looks good, this fixes the python3 incompatibility traceback. Could also turn those filters into list comprehensions.
I notice that this enables another bug on [line 437](https://github.com/ansible/ansible/pull/27318/files#diff-18b06d7d80bc10ddff5b4aed6edc379cL437). In Python3 that filter item is always found to be True on [line 443](https://github.com/ansible/ansible/pull/27318/files#diff-18b06d7d80bc10ddff5b4aed6edc379cL443) leading to a perpetual changed=true. Would you mind adding a fix there too? 437 could either be wrapped in list() or to make it easier to to understand could also do something like `security_groups_to_update = [sid for sid in intersection if targets[sid].get('SecurityGroups') and current_targets[sid]['SecurityGroups'] != targets[sid]['SecurityGroups']]`. After making that a list you can remove line 443 and just do the iteration over it.